avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -842,8 +844,18 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
         } else {
           // Special record name indicates format
           List<org.apache.avro.Schema.Field> fields = new ArrayList<>();
-          final org.apache.avro.Schema mapSchema = org.apache.avro.Schema.createRecord(
-              MAP_ENTRY_TYPE_NAME, null, namespace, false);
+          final org.apache.avro.Schema mapSchema;
+          if (schema.name() == null) {
+            mapSchema = org.apache.avro.Schema.createRecord(
+                MAP_ENTRY_TYPE_NAME,
+                null,
+                namespace,
+                false
+            );
+          } else {
+            mapSchema = org.apache.avro.Schema.createRecord(name, null, namespace, false);
+            mapSchema.addProp(CONNECT_NAMED_MAP_PROP, CONNECT_NAMED_MAP_PROP_TRUE);
```

Comment:
should the property be `connect.internal.type: MapEntry`?

Reply:
yeah that works - I'll use that instead of the named map thing